src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/README.md

```diff
@@ -18,6 +23,100 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage
+
+### Enhanced Logging Capabilities
+
+The package includes a custom logging generator that enhances the default .NET logging capabilities by replacing the default generator. This generator automatically logs the contents of collections and offers advanced logging features, significantly improving the debugging and monitoring process.
+
+```csharp
+[LoggerMessage(1, LogLevel.Information, "These are the contents of my dictionary: {dictionary}")]
+internal static partial void LogMyDictionary(ILogger<Program> logger, Dictionary<int, string> temperature);
+```
+
+It also adds the `LogProperties` attribute which can be applied to a an object parameter of a `LoggerMessage` method. It introspects the passed-in object and automatically adds tags for all its properties. This leads to more informative logs without the need for manual tagging of each property.
+
+```csharp
+[LoggerMessage(1, LogLevel.Information, "Detected a new temperature: {temperature}")]
```

Comment:
```suggestion
[LoggerMessage(1, LogLevel.Information, "Detected a new temperature: {Temperature}")]
```

Reply:
This should match the parameter name, which is lower case.

Reply:
Examples in extensions

https://github.com/dotnet/extensions/blob/release/8.0/src/Libraries/Microsoft.Extensions.Hosting.Testing/HostTerminatorService.cs#L63-L64

https://github.com/dotnet/extensions/blob/release/8.0/src/Libraries/Microsoft.AspNetCore.HeaderParsing/HeaderParsingFeature.cs#L201-L209

And the doc: https://learn.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator#basic-usage

Reply:
Keep in mind that this is using a different generator which adds additional constraints and diagnostics (this is using Microsoft.Extensions.Telemetry.Abstractions logging generator)

I believe in the past I've seen warnings when parameter name doesn't exactly match the template, @geeknoid  can you confirm?

Reply:
I checked back and it does seem like this is not case-sensitive as you suggested, which also seems to be called out explicitly https://learn.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator#case-insensitive-template-name-support . Anyway, since it's not a convention that we should upper case those, I've left it lowercase for now (I also hope it helps describing that what should match is paramter name as opposed to the parameter type, since in my example both are the same).

Reply:
First letter can be either lowercased or uppercased, you can check Iliar's PR: #4584